source/tracing_context_impl.cc

```diff
@@ -138,7 +138,7 @@ TracingContextImpl::TracingContextImpl(
     SpanContextExtensionPtr parent_ext_span_context, RandomGenerator& random)
     : parent_span_context_(std::move(parent_span_context)),
       parent_ext_span_context_(std::move(parent_ext_span_context)) {
-  segment_store_.set_traceid(random.uuid());
+  segment_store_.set_traceid(parent_span_context_->traceId());
```

Comment:
It should be parent trace ID, if not exist parent trace ID, create a UUID?

Reply:
Thanks. But may be we should check if the parent context is existed or not?

```
segment_store_.set_traceid(parent_span_context_ ? parent_span_context_->traceId() : random.uuid());
```